pymeasure/instruments/agilent/agilent34970A.py

```diff
@@ -0,0 +1,58 @@
+#
+# this file is intended to become part of the PyMeasure package
+#
+# agilent34970A.py
+# module to support the HP/Agilent/Keysight 34970A multimeter
+# Chris Freitag, August 20, 2019
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strickt_discrete_set,\
+    strict_range, joined_validators
+from time import time
+from pyvisa.errors import VisaIOError
+
+# Capitalize string arguments to allow for better conformity with other WFG's
+def capitalize_string(string: str, *args, **kwargs):
+    return string.upper()
+
+
+# Combine the capitalize function and validator
+string_validator = joined_validators(capitalize_string, strict_discrete_set)
```

Comment:
I am not sure, this actually works the way you intend it to work... see also here: https://github.com/ralph-group/pymeasure/pull/209#discussion_r306018070